compiler/rustc_privacy/src/lib.rs

```diff
@@ -409,8 +405,24 @@ impl VisibilityLike for ty::Visibility {
     }
 }
 
-impl VisibilityLike for Option<EffectiveVisibility> {
-    const MAX: Self = Some(EffectiveVisibility::from_vis(ty::Visibility::Public));
+struct FindMinEffectiveVis(EffectiveVisibility);
```

Comment:
Could you actually make two wrapper structures
```rust
struct ShallowEffectiveVis(EffectiveVisibility);
struct NonShallowEffectiveVis(EffectiveVisibility);
```
so it's immediately clear what's the difference?